src/components/Comment/index.spec.tsx

```diff
@@ -1,82 +1,309 @@
+    const addonId = 222;
+    const fileName = 'manifest.json';
+    const line = 432;
+    const versionId = 321;
+    const store = configureStore();
+    const dispatchSpy = spyOn(store, 'dispatch');
+
+    const fakeThunk = createFakeThunk();
+    const _manageComment = fakeThunk.createThunk;
+
+    const commentText = 'Example of a comment';
+
+    const root = render({
+      _manageComment,
+      addonId,
+      fileName,
+      line,
+      store,
+      versionId,
+    });
+
+    root.find(`.${styles.textarea}`).simulate(
+      'change',
+      createFakeChangeEvent({
+        name: 'input',
+        value: commentText,
+      }),
     );
+
+    root.find(`.${styles.form}`).simulate('submit', createFakeEvent());
+
+    expect(dispatchSpy).toHaveBeenCalledWith(fakeThunk.thunk);
+    expect(_manageComment).toHaveBeenCalledWith({
+      addonId,
+      cannedResponseId: undefined,
+      comment: commentText,
+      commentId: undefined,
+      fileName,
+      line,
+      versionId,
+    });
   });
 
-  describe('keydown protection', () => {
-    const createFakeTextareaRef = () => {
-      return {
-        ...React.createRef(),
-        current: {
-          ...document.createElement('textarea'),
-          addEventListener: jest.fn(),
-          removeEventListener: jest.fn(),
-        },
-      };
-    };
+  it('renders a pending state while saving a comment', () => {
+    const keyParams = { fileName: null, line: null, versionId: 1 };
+    const pendingCommentText = 'Example of a comment being edited';
+    const store = configureStore();
+    store.dispatch(
+      actions.beginSaveComment({ ...keyParams, pendingCommentText }),
+    );
+
+    const root = render({ readOnly: false, store, ...keyParams });
 
+    const textarea = root.find(`.${styles.textarea}`);
+    expect(textarea).toHaveProp('disabled', true);
+    expect(textarea).toHaveProp('value', pendingCommentText);
+
+    const button = root.find(Button);
+    expect(button).toHaveProp('disabled', true);
+    expect(button).toHaveText('Saving…');
+  });
+
+  it('seeds the form with an initial comment', () => {
+    const commentId = 1;
+    const commentText = 'Example of a previously saved comment';
+    const comment = createFakeExternalComment({
+      id: commentId,
+      comment: commentText,
+    });
+    const { store } = dispatchComment({ comment });
+    const dispatchSpy = spyOn(store, 'dispatch');
+
+    const fakeThunk = createFakeThunk();
+    const _manageComment = fakeThunk.createThunk;
+
+    const root = render({
+      _manageComment,
+      commentId,
+      store,
+    });
+
+    root.find(`.${styles.form}`).simulate('submit', createFakeEvent());
+
+    expect(dispatchSpy).toHaveBeenCalledWith(fakeThunk.thunk);
+    expect(_manageComment).toHaveBeenCalledWith(
+      expect.objectContaining({
+        comment: commentText,
+        commentId: comment.id,
+      }),
+    );
+  });
+
+  it('seeds the form with initialCommentText', () => {
```

Comment:
Also, I don't see a test where both existing comment is dispatched, and also `beginSaveComment` is dispatched with a value for `pendingCommentText`. I.e., the case where a form is opened for an existing comment, and then it is submitted with a changed value. It seems like that case should be covered.

Reply:
The "it lets you update a previously saved comment" [test](https://github.com/mozilla/addons-code-manager/pull/1045/files#diff-ae5df7d4b15cf0f3adbcf02ae01c5e85R259) implies that the situation you described will work. However, I have since updated the "it seeds..." tests which I think makes it more clear.